tests/functional/models/test_firefox_new_tab_metrics_factory.py

```diff
@@ -0,0 +1,121 @@
+from botocore.exceptions import BotoCoreError
+
+import app.config
+from app.models.metrics.firefox_new_tab_metrics_factory import FirefoxNewTabMetricsFactory
+
+
+@pytest.mark.asyncio  # This pytest-asyncio decorator allows us to use an async side_effect
+class TestFirefoxNewTabMetricsFactory:
+
+    SLATE_ID = 'f99178fb-6bd0-4fa1-8109-cda181b697f6'  # Matches the slate_id in firefox_new_tab_engagement.json
+    feature_group_name: str = FirefoxNewTabMetricsFactory.get_feature_group_name()
+
+    def setup(self):
+        global_sdk_config.set_sdk_enabled(False)
+        records = self._read_json_asset('firefox_new_tab_engagement.json')
+        # Map id to record (use record[0] assuming that ID is the first feature in the above json fixture)
+        self._firefox_new_tab_engagement_record_by_id = {record[0]['ValueAsString']: record for record in records}
+
+    async def _get_mocked_feature_store_client(self, monkeypatch):
+        # Mock boto3. Localstack currently does not support Feature Group, so we need to mock it ourselves.
+        client = MagicMock()
+        aioboto3 = MagicMock(client=client)
+        monkeypatch.setattr("app.models.metrics.firefox_new_tab_metrics_factory.aioboto3", aioboto3)
+        # Mock return values for 'get_record' calls.
+        # Use partial pass in 'self' so it can access attributes loaded during setup.
+        client.return_value.__aenter__.return_value.get_record.side_effect = \
+            partial(self._feature_store_get_record_stub, self)
+        return client
+
+    def _read_json_asset(self, filename: str):
+        with open(os.path.join(app.config.ROOT_DIR, 'tests/assets/json/', filename)) as f:
+            return json.load(f)
+
+    def _feature_store_get_record_stub(self, *args, **kwargs):
+        """
+        This function simulates the FeatureStore get_record operation.
+        :param args:
+        :param kwargs: get_record arguments
+        :return: simplified version of get_record return value: only the record is returned (not any request metadata)
+        """
+        record_identifier_value_as_string = kwargs['RecordIdentifierValueAsString']
+        feature_group_name = kwargs['FeatureGroupName']
+        feature_names = kwargs['FeatureNames']
+
+        if feature_group_name != self.feature_group_name:
+            # Raise an exception if the feature group is not found.
+            # In reality FeatureStore raises a ClientError, but the BotoCoreError is easier to construct for testing.
+            raise BotoCoreError()
+
+        full_record = self._firefox_new_tab_engagement_record_by_id.get(record_identifier_value_as_string, {})
+        # Only return requested features to check that a valid model can be constructed from the requested features.
+        filtered_record = [feature for feature in full_record if feature['FeatureName'] in feature_names]
+        return {'Record': filtered_record}
+
+    async def test_get_existing_records(self, monkeypatch):
```

Comment:
When I try to run the tests, they're getting skipped in my editor and I get a warning about not having a suitable plugin installed (I looked at my pip environment and pytest-asyncio was there). I didn't spend time trying to get them running tonight, but wanted to flag. Is there another step needed here to get these running, a run configuration or something?

Reply:
Did you run the tests using docker-compose? [The development setup docs](https://github.com/Pocket/recommendation-api/blob/main/docs/development.md#runningdebugging-app-tests-in-pycharm) describe how to run the tests.

Reply:
I did use that setup. It turns out I gotta do `build` and `up` in the Pycharm terminal—the regular terminal doesn't work for this, which is where I normally do command line stuff.